packages/dara-core/dara/core/internal/routing.py

```diff
@@ -361,8 +361,13 @@ async def upload(
             content = cast(bytes, await data.read())
 
             if resolver_id is not None:
-                resolver = action_registry.get(resolver_id)
+                registry_mgr: RegistryLookup = utils_registry.get('RegistryLookup')
+                resolver = await registry_mgr.get(action_registry, resolver_id)
                 content = resolver(content, data.filename)
```

Comment:
We have a utility for running user-defined functions - `dara.core.internal.utils.run_user_handler` which handles async functions etc, let's use it here instead of inspect etc